# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/canvas/ClientWebGLContext.cpp
# Commit: 073302d26c5e
# Full Hash: 073302d26c5e1ac8229fca1b32656de68756e0f2
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2020-06-11 09:34:54
# Regressor Bug: 1632249
# File Overlap Count: 1
# Description:
#   Bug 1632249 - Create a separate WebGLContext::PresentInto function for WebXR. r=jgilbert
#   
#   - Do not apply mIsCanvasDirty bailout when presenting from a opaque FBO
#   back buffer.
#   - Create a separate WebGLContext::PresentInto with more specific XR
# ==============================================================================

diff -r 7c94d37c446e -r 073302d26c5e dom/canvas/ClientWebGLContext.cpp
--- a/dom/canvas/ClientWebGLContext.cpp	Thu Jun 11 06:37:36 2020 +0000
+++ b/dom/canvas/ClientWebGLContext.cpp	Thu Jun 11 06:37:38 2020 +0000
@@ -419,7 +419,7 @@
 
 void ClientWebGLContext::Present(WebGLFramebufferJS* const fb,
                                  const layers::TextureType type) {
-  if (!mIsCanvasDirty) return;
+  if (!mIsCanvasDirty && !fb) return;
   mIsCanvasDirty = false;
   mFrontBufferSnapshot = nullptr;
 